snoop/data/views.py

```diff
@@ -312,3 +312,14 @@ def destroy(self, request, pk=None, **kwargs):
         """Collection-aware overload that also checks permission to write tag."""
         self.check_ownership(pk)
         return super().destroy(request, pk, **kwargs)
+
+
+@collection_view
+def thumbnail(request):
+    digest_sha3 = request.GET['digest']
+    size = request.GET['size']
+    digest_id = models.Digest.objects.get(blob_id=digest_sha3).pk
+    thumbnail_obj = models.Thumbnail.objects.get(original=digest_id, size=size)
```

Comment:
you can collapse the two DB queries into one: `models.Thumbnail.objects.get(size=size, original__pk=digest_sha3)`